test/js-native-api/6_object_wrap/nested_wrap.cc

```diff
@@ -0,0 +1,99 @@
+#include "nested_wrap.h"
+#include "../common.h"
+#include "../entry_point.h"
+
+napi_ref NestedWrap::constructor;
+static int finalization_count = 0;
+
+NestedWrap::NestedWrap() : env_(nullptr), wrapper_(nullptr), nested_(nullptr) {}
```

Comment:
nit: in modern C++ it is preferable to default initialize fields directly in the class. E.g.
```
  napi_env env_{};
  napi_ref wrapper_{};
  napi_ref nested_{};
};
```
If a constructor provides different initialization for the fields, then these declarations are going to be ignored.